spring-kafka/src/main/java/org/springframework/kafka/listener/FallbackBatchErrorHandler.java

```diff
@@ -122,4 +141,23 @@ public void onPartitionsAssigned(Consumer<?, ?> consumer, Collection<TopicPartit
 		}
 	}
 
+	private final class SeekAfterRecoverFailsOrInterrupted implements CommonErrorHandler {
```

Comment:
Don't we need to have a package protected ctor to avoid extra generated code around constructor ?
Or new Java does that optimally already?

Reply:
Good catch; thanks.